common/src/main/java/com/linecorp/centraldogma/common/Entry.java

```diff
@@ -174,101 +165,58 @@ public void ifHasContent(Consumer<? super T> consumer) {
     }
 
     /**
-     * Returns the content of this {@link Entry}.
+     * Returns a new {@link Entry} which applied the specified {@link Query} to the {@link #content()}.
      *
      * @throws IllegalStateException if this {@link Entry} is a directory
-     */
+     * @throws QuerySyntaxException if the syntax of specified {@link Query} is invalid
+     * @throws QueryExecutionException if an {@link Exception} is raised while applying the specified
+     *                                 {@link Query} to the {@link #content()}
+     */
+    public Entry<T> applyQuery(Query<T> query) {
```

Comment:
Could you explain why you had to introduce this method instead of using `Query.apply()`?